test/metabase/api/database_test.clj

```diff
@@ -281,6 +281,18 @@
            (let [resp ((mt/user->client :rasta) :get 200 (format "database/%d/metadata" (mt/id)))]
              (assoc resp :tables (filter #(= "CATEGORIES" (:name %)) (:tables resp))))))))
 
+(deftest fetch-database-metadata-without-hidden-test
+  (testing "GET /api/database/:id/metadata"
+    (try
+      (db/update! Table (mt/id :categories) :visibility_type "hidden")
+      (is (= ["CHECKINS" "USERS" "VENUES"]
+             (->> ((mt/user->client :rasta) :get 200 (format "database/%d/metadata" (mt/id)))
+                  :tables
+                  (map :name))))
+      (finally
+        (db/update! Table (mt/id :categories) :visibility_type nil)))))
```

Comment:
@camsaul I need to add more tests, but wanted to check in on the right approach first. Is there a better way to temporarily adjust the fixture data like I do here setting `:visibility_type` to hidden?

Reply:
You can use `with-temp-vals-in-dB` to do it. Example: https://github.com/metabase/metabase/blob/005b4567f8ab8eb05d1719e98ded66441332bb49/modules/drivers/bigquery/test/metabase/driver/bigquery/query_processor_test.clj#L126